include/ndarray.hpp

```diff
@@ -94,8 +98,11 @@ class NDArray {
   //==========================================================================
   // Constant Methods
 
+  // Return underlying data vector
+  const std::vector<T>& data() const;
```

Comment:
I think having a way to return a reference to the underlying container is a great idea, though I think a different name should be used. When I see `data`, I would expect to get a pointer back, like with the STL containers. Perhaps we could rename this to be `data_vector` ? If you have another idea, I am very open to suggestions. After I merge your PR, I will probably add a `data` method which returns a pointer to the data to make it a little more STL-like.

Reply:
The name `data_vector` seems just fine. I have changed my commit from 4ffd26e to bf392fc.
diff: https://github.com/munehiro-k/ndarray/compare/4ffd26e..bf392fc

The idea to make `data` member function to return a pointer sounds also good. I have missed to consider to be STL-like.